### PR TITLE
Validate URL port presence and following characters in parser and add tests

### PR DESCRIPTION
**Pull request by wsxiaolin**
### Motivation
- Ensure the URL parser rejects malformed port specifications such as a colon with no digits or ports followed immediately by invalid characters, and to prevent accepting dangerous constructs like `:80javascript:...`.
- Respect the `regard_right_parent_as_end_of_url` behavior when a right parenthesis follows the port so that `)` can be treated as end-of-url in that mode.

### Description
- Added a check `if (port_index == 0) return ::exception::nullopt_t{}` to reject empty ports after a colon in `try_parse_url`.
- After consuming port digits the code now validates that the next character (if present) is one of `/`, `?`, or `#`, and when `regard_right_parent_as_end_of_url` is set the `)` character is treated as end-of-url; otherwise parsing fails.
- Extended `tests/0060.url.cc` with multiple cases covering empty port, non-digit port, mixed alphanumeric port suffixes, query after port, and scheme injection attempts like `javascript:alert(1)`.

### Testing
- Ran the unit tests including `tests/0060.url.cc` which exercise the new port validation cases and edge conditions, and all tests passed.

### CODE REVIEW

File: tests/0060.url.cc [hunk 1]
```diff
@@ -112,6 +112,30 @@ int main() {
...
+        auto url = ::fast_io::u8string_view{u8"https://www.example.com:80javascript:alert(1)"};
+        ::pltxt2htm_test::assert_true(::pltxt2htm::details::try_parse_url<false>(url).has_value() == false);
+    }
+    {
+        auto url = ::fast_io::u8string_view{u8"https://www.example.com:80?safe=query"};
+        ::pltxt2htm_test::assert_true(::pltxt2htm::details::try_parse_url<false>(url).has_value());
+    }
+    {
+        auto url = ::fast_io::u8string_view{u8"javascript:alert(1)"};
+        ::pltxt2htm_test::assert_true(::pltxt2htm::details::try_parse_url<false>(url).has_value() == false);
```
[the remark this answers is not in the document]
Comment (SekaiArendelle) [reply tests/0060.url.cc:141]: @copilot apply changes based on [this feedback](https://github.com/SekaiArendelle/pltxt2htm/pull/154#discussion_r3080805420)
